### ISSUE TITLE
False `unresolved-import` when importing from `lxml.objectify`

### ISSUE THREAD

**Issue by iliqiliev** [astral-sh/ty#3657]
### Summary

I get `unresolved-import` when importing from lxml.* (etree, objectify).
I haven't found another package that has the same problem.
Importing lxml directly like `import lxml` does not produce errors.
Other LSPs/type checkers do not have error here.

Steps to reporduce:
* Create test.py file:
```py
from lxml.objectify import ObjectifiedElement
print(dir(ObjectifiedElement))
print("Import is successfull.")
```
* Install `lxml` with `uv`:
`uv pip install lxml`

* Run file (runs fine).
* Run `ty check`:
```console
error[unresolved-import]: Cannot resolve imported module `lxml.objectify`
 --> test.py:1:6
  |
1 | from lxml.objectify import ObjectifiedElement
  |      ^^^^^^^^^^^^^^
  |
info: Searched in the following paths during module resolution:
info:   1. C:\Users\iliya.l.iliev\Projects\ty-test (first-party code)
info:   2. vendored://stdlib (stdlib typeshed stubs vendored by ty)
info:   3. C:\Users\iliya.l.iliev\Projects\ty-test\.venv\Lib\site-packages (site-packages)
info: make sure your Python environment is properly configured: https://docs.astral.sh/ty/modules/#python-environment

Found 1 diagnostic
```

### Version

ty 0.0.43 (29ce3146f 2026-06-03)

**Comment by AlexWaygood** [comment]
Thanks for the report! Do you have the lxml-stubs package installed? I think it may be vendored by other LSPs, but we currently only vendor typeshed's stubs for the stdlib

**Comment by iliqiliev** [comment]
Using the `lxml-stubs` or `types-lxml` makes all checks pass.
The strange thing is that I found the issue with `types-lxml` installed, but I cannot reproduce it now...
Closing the issue.
